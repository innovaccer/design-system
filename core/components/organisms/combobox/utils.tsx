import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  inputTriggerRef?: any,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>,
  multiSelect?: boolean,
  listRef?: any
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      navigateOptions('up', focusedOption, setFocusedOption, listRef);
      break;
    case 'ArrowDown':
      event.preventDefault();
      navigateOptions('down', focusedOption, setFocusedOption, listRef);
      break;
    case 'Enter':
      handleEnterKey(focusedOption, multiSelect, inputTriggerRef, listRef, setFocusedOption);
      setHighlightLastItem?.(false);
      setHighlightFirstItem?.(false);
      break;
    case 'Escape':
      setOpenPopover?.(false);
      inputTriggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }
};

const handleEnterKey = (
  focusedOption: Element | undefined,
  multiSelect?: boolean,
  inputTriggerRef?: any,
  listRef?: any,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>
) => {
  (focusedOption as HTMLElement)?.click();
  if (!multiSelect) {
    inputTriggerRef.current.focus();
  } else {
    // to focus first option by default when last option is selected
    const listItems = listRef.current.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
    const listArr = Array.from(listItems);
    const index = listArr.findIndex((item) => {
      return item == focusedOption;
    });

    if (index === listArr.length - 1) {
      (listItems[0] as HTMLElement).focus();
      setFocusedOption && setFocusedOption(listItems[0]);
      listItems[0].scrollIntoView({ block: 'center' });
    }
  }
};

const navigateOptions = (
  direction: string,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any
) => {
  const listItems = listRef.current.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let index = Array.from(listItems).findIndex((item) => {
    return item == focusedOption;
  });

  if (index === -1) {
    index = direction === 'up' ? listItems.length - 1 : 0;
  } else {
    index = direction === 'up' ? (index - 1 + listItems.length) % listItems.length : (index + 1) % listItems.length;
  }

  const targetOption = listItems[index];

  (targetOption as HTMLElement).focus();
  setFocusedOption && setFocusedOption(targetOption);
  targetOption?.scrollIntoView?.({ block: 'center' });
};
