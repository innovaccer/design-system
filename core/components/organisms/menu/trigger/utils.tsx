import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean | undefined>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      setOpenPopover?.(true);
      setHighlightLastItem?.(true);
      break;
    case 'ArrowDown':
      event.preventDefault();
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'Escape':
    case 'Tab':
      setOpenPopover?.(false);
      break;
    default:
      break;
  }
};

export const focusListItem = (
  position: string,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any
) => {
  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  if (position === 'down') {
    targetOption = listItems?.[0];
  } else {
    targetOption = listItems?.[listItems.length - 1];
  }
  (targetOption as HTMLElement)?.focus();

  if (targetOption && typeof targetOption.scrollIntoView === 'function') {
    (targetOption as HTMLElement)?.scrollIntoView({ block: 'end' });
  }
  setFocusedOption && setFocusedOption(targetOption);
};
