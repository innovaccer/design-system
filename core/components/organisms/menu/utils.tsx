import React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean | undefined>>,
  menuTriggerRef?: React.RefObject<HTMLButtonElement | null>,
  listRef?: React.RefObject<HTMLDivElement | null>,
  subListRef?: React.RefObject<HTMLDivElement | null> | null,
  isSubMenuTrigger?: boolean,
  triggerRef?: React.RefObject<HTMLDivElement | null> | React.MutableRefObject<HTMLDivElement | null>,
  menuID?: string,
  triggerID?: string,
  parentListRef?: React.RefObject<HTMLDivElement | null> | null
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      navigateOptions('up', focusedOption, setFocusedOption, listRef);
      break;
    case 'ArrowDown':
      event.preventDefault();
      navigateOptions('down', focusedOption, setFocusedOption, listRef);
      break;
    case 'Enter':
      (focusedOption as HTMLElement)?.click();
      setOpenPopover?.(false);
      break;
    case 'Escape':
      setOpenPopover?.(false);
      if (triggerRef && !isSubMenuTrigger) {
        triggerRef?.current?.focus();
      } else {
        menuTriggerRef?.current?.focus();
      }
      setFocusedOption?.(undefined);
      break;
    case 'Tab':
      setOpenPopover?.(false);
      break;
    case 'ArrowRight':
      navigateSubMenu(isSubMenuTrigger, 'right', subListRef, menuID, triggerID, parentListRef);
      break;
    case 'ArrowLeft':
      navigateSubMenu(isSubMenuTrigger, 'left', subListRef, menuID, triggerID, parentListRef);
      break;
    default:
      break;
  }
};

const navigateOptions = (
  direction: string,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any
) => {
  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let index = Array.from(listItems).findIndex((item) => {
    return item == focusedOption;
  });

  if (index === -1) {
    index = direction === 'up' ? listItems.length - 1 : 0;
  } else {
    index = direction === 'up' ? (index - 1 + listItems.length) % listItems.length : (index + 1) % listItems.length;
  }

  const targetOption = listItems[index];
  (targetOption as HTMLElement).focus();
  setFocusedOption && setFocusedOption(targetOption);
  targetOption?.scrollIntoView?.({ block: 'center' });
};

const navigateSubMenu = (
  isSubMenuTrigger?: boolean,
  direction?: string,
  subListRef?: React.RefObject<HTMLDivElement | null> | null,
  menuID?: string,
  triggerID?: string,
  parentListRef?: React.RefObject<HTMLDivElement | null> | null
) => {
  const element = document.querySelector(`[data-name="${menuID}"]`);
  const menuPlacement = element?.getAttribute('data-placement');

  if (isSubMenuTrigger) {
    if (
      (direction === 'right' && menuPlacement?.includes('right')) ||
      (direction === 'left' && menuPlacement?.includes('left'))
    ) {
      const listItems = subListRef?.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
      (listItems?.[0] as HTMLElement).focus();
    }
  } else if (
    (direction === 'left' && menuPlacement?.includes('right')) ||
    (direction === 'right' && menuPlacement?.includes('left'))
  ) {
    const triggerElement = parentListRef?.current?.querySelector(`#${triggerID}`)?.firstChild;
    (triggerElement as HTMLElement)?.focus();
  }
};
