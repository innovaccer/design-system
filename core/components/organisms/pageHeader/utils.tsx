import * as React from 'react';
import classNames from 'classnames';
import { navigationPositionType } from './PageHeader';
import { Heading, Column } from '@/index';
import styles from '@css/components/pageHeader.module.css';

export const Status = (props: {
  status: React.ReactNode;
  meta: React.ReactNode;
  navigationPosition: navigationPositionType;
  navigation: React.ReactNode;
  tabs: React.ReactNode;
}) => {
  const { status, meta, navigationPosition, navigation, tabs } = props;

  const statusClasses = classNames({
    [styles['PageHeader-statusWrapper']]: true,
    'mb-3': (navigationPosition === 'bottom' && navigation) || tabs,
  });

  return (
    <>
      {(status || meta) && (
        <div className={statusClasses} data-test="DesignSystem-PageHeader--Status">
          {status}
          {meta}
        </div>
      )}
    </>
  );
};

export const Action = (props: { actions: React.ReactNode; navigation: React.ReactNode; stepper: React.ReactNode }) => {
  const { actions, navigation, stepper } = props;

  return (
    <>
      {actions ? (
        <Column size="4" sizeXL="4" sizeM="4" data-test="DesignSystem-PageHeader--Actions">
          <div className={styles['PageHeader-actionsWrapper']}>{actions}</div>
        </Column>
      ) : (
        (navigation || stepper) && (
          <Column size="4" sizeXL="4" sizeM="4" data-test="DesignSystem-PageHeader--Actions">
            <div className={styles['PageHeader-actionsWrapper']}></div>
          </Column>
        )
      )}
    </>
  );
};

export const Nav = (props: { navigation: React.ReactNode; stepper: React.ReactNode }) => {
  const { navigation, stepper } = props;

  if (!navigation && !stepper) {
    return null;
  }
  return (
    <div className={styles['PageHeader-navigationWrapper']} data-test="DesignSystem-PageHeader--Nav">
      {navigation || stepper}
    </div>
  );
};

export const CenterNav = (props: {
  colSize: string;
  breadcrumbs: React.ReactNode;
  navigationPosition: navigationPositionType;
  navigation: React.ReactNode;
  stepper: React.ReactNode;
}) => {
  const { colSize, breadcrumbs, navigationPosition } = props;
  return (
    <>
      {(!breadcrumbs || navigationPosition === 'center') && colSize === '4' && (
        <Column size="4" sizeXL="4" sizeM="4" data-test="DesignSystem-PageHeader--CenterNav">
          <Nav {...props} />
        </Column>
      )}
    </>
  );
};

export const BackButton = (props: { button: React.ReactNode }) => {
  const { button } = props;
  return (
    <>
      {button && (
        <div className="mr-5 my-3" data-test="DesignSystem-PageHeader--Button">
          {button}
        </div>
      )}
    </>
  );
};

export const Title = (props: { badge: React.ReactNode; title: string }) => {
  const { badge, title } = props;
  return (
    <div className={styles['PageHeader-titleWrapper']} data-test="DesignSystem-PageHeader--Title">
      <Heading className={styles['PageHeader-title']}>{title}</Heading>
      {badge}
    </div>
  );
};
