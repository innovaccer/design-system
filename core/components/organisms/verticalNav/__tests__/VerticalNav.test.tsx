import * as React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { VerticalNavProps as Props } from '@/index.type';
import { VerticalNav } from '@/index';
import { testHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/testHelper';

const onClick = jest.fn();
const onToggle = jest.fn();
const BooleanValue = [true, false];
const active = {
  name: 'care_management.timeline',
};

const menus = [
  {
    name: 'patient_360',
    label: 'Patient 360',
    icon: 'assignment_ind',
    link: '/patient360',
    count: 10,
    group: 'Section 1',
  },
  {
    name: 'care_management',
    label: 'Care Management and Resources',
    icon: 'forum',
    group: 'Section 2',
    subMenu: [
      {
        name: 'care_management.timeline',
        label: 'Timeline',
        icon: 'assignment_ind',
      },
      {
        name: 'care_management.care_plans',
        label: 'Care Plans',
        icon: 'events',
      },
    ],
  },
  {
    name: 'episodes',
    label: 'Episodes',
    disabled: true,
    icon: 'airline_seat_flat_angled',
    group: 'Section 2',
    count: 100,
  },
  {
    name: 'risk',
    label: 'Risk',
    icon: 'favorite',
    group: 'Section 2',
    subMenu: [
      {
        name: 'risk.timeline',
        label: 'Timeline',
      },
      {
        name: 'risk.care_plans',
        label: 'Care Plans',
      },
    ],
  },
];

const menusWithExpandedSubMenu = [
  {
    name: 'patient_360',
    label: 'Patient 360',
    icon: 'assignment_ind',
    link: '/patient360',
    count: 10,
    group: 'Section 1',
  },
  {
    name: 'care_management',
    label: 'Care Management and Resources',
    icon: 'forum',
    group: 'Section 2',
    expanded: true,
    subMenu: [
      {
        name: 'care_management.timeline',
        label: 'Timeline',
        icon: 'assignment_ind',
      },
    ],
  },
  {
    name: 'episodes',
    label: 'Episodes',
    disabled: true,
    icon: 'airline_seat_flat_angled',
    group: 'Section 2',
    count: 100,
  },
];

describe('Vertical Navigation component', () => {
  const mapper = {
    expanded: valueHelper(true, { required: true }),
    rounded: valueHelper(BooleanValue, { required: true }),
    menus: valueHelper(menus, { required: true }),
    active: valueHelper(active, { required: true }),
    onClick: valueHelper(onClick, { required: true }),
    onToggle: valueHelper(onToggle, { required: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<VerticalNav {...attr} />);

      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Vertical Navigation component', () => {
  const mapper = {
    expanded: valueHelper(false, { required: true }),
    menus: valueHelper(menus, { required: true }),
    active: valueHelper(active, { required: true }),
    onClick: valueHelper(onClick, { required: true }),
    onToggle: valueHelper(onToggle, { required: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<VerticalNav {...attr} />);

      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Vertical Navigation component with prop: menus', () => {
  const disabledIndex = 2;

  it('renders menus', () => {
    const { getAllByTestId } = render(<VerticalNav menus={menus} />);
    expect(getAllByTestId('DesignSystem-VerticalNav--Item')).toHaveLength(menus.length);
  });

  it('renders icon of parent items in collapsed state ', () => {
    const currActive = { name: 'patient_360' };
    const { getAllByTestId } = render(<VerticalNav menus={menus} expanded={false} active={currActive} />);

    expect(getAllByTestId('DesignSystem-VerticalNav--Icon')).toHaveLength(menus.length);
  });

  it('renders icon of child items in collapsed state if the parent was initially expanded', () => {
    const childIconCount = 2;

    const { getAllByTestId } = render(<VerticalNav menus={menus} active={active} />);

    expect(getAllByTestId('DesignSystem-VerticalNav--Icon')).toHaveLength(menus.length + childIconCount);
  });

  it('renders menus with count', () => {
    const { getAllByTestId } = render(<VerticalNav menus={menus} expanded={true} />);
    expect(getAllByTestId('DesignSystem-VerticalNav--Pills')[0]).toHaveTextContent('10');
    expect(getAllByTestId('DesignSystem-VerticalNav--Pills')[1]).toHaveTextContent('99+');
  });

  it('renders menus with sections', () => {
    const { getAllByTestId } = render(<VerticalNav menus={menus} expanded={true} />);

    const sections = getAllByTestId('DesignSystem-VerticalNav--Section');
    expect(sections).toHaveLength(2);
    expect(sections[0]).toHaveTextContent('Section 1');
    expect(sections[1]).toHaveTextContent('Section 2');
  });

  it('renders menus with disabled menu', () => {
    const { getAllByTestId } = render(<VerticalNav menus={menus} expanded={true} active={{ name: 'patient_360' }} />);

    const disabledMenuItem = getAllByTestId('DesignSystem-VerticalNav--Item')[disabledIndex];
    expect(disabledMenuItem).toHaveClass('color-inverse-lightest');

    const disabledText = getAllByTestId('DesignSystem-VerticalNav--Text')[disabledIndex];
    expect(disabledText).toHaveClass('color-inverse-lightest');

    const disabledIcon = getAllByTestId('DesignSystem-VerticalNav--Icon')[disabledIndex];
    expect(disabledIcon).toHaveStyle('color: var(--inverse-lightest)');

    const disabledPill = getAllByTestId('DesignSystem-VerticalNav--Pills')[1];
    expect(disabledPill).toHaveClass('Badge--subtle-secondary');
  });

  it('renders child items if parent item is clicked and panel is expanded', () => {
    const menuClicked = 1;
    const { getAllByTestId } = render(
      <VerticalNav expanded={true} menus={menus} active={{ name: 'patient_360' }} onClick={onClick} />
    );

    const activeMenu = getAllByTestId('DesignSystem-VerticalNav--Item')[menuClicked];
    fireEvent.click(activeMenu);

    const childItems = getAllByTestId('DesignSystem-VerticalNav--Text');
    expect(childItems[2].textContent).toMatch('Timeline');
    expect(childItems[3].textContent).toMatch('Care Plans');
  });
});

describe('Vertical Navigation component active menu', () => {
  const activeIndex = 0;

  it('renders menus with active menu', () => {
    const currActive = { name: 'patient_360' };
    const { getAllByTestId } = render(<VerticalNav expanded={true} menus={menus} active={currActive} />);

    const activeMenuItem = getAllByTestId('DesignSystem-VerticalNav--Item')[activeIndex];
    expect(activeMenuItem).toHaveClass('color-primary-dark');

    const activeText = getAllByTestId('DesignSystem-VerticalNav--Text')[activeIndex];
    expect(activeText).toHaveClass('color-primary-dark');

    const activeIcon = getAllByTestId('DesignSystem-VerticalNav--Icon')[activeIndex];
    expect(activeIcon).toHaveStyle('color: var(--primary-dark)');

    const activePill = getAllByTestId('DesignSystem-VerticalNav--Pills')[activeIndex];
    expect(activePill).toHaveClass('Badge--primary');
  });

  it('renders active menu with link', () => {
    const { getAllByTestId } = render(<VerticalNav expanded={true} menus={menus} active={{ link: '/patient360' }} />);
    const activeMenu = getAllByTestId('DesignSystem-VerticalNav--Text')[activeIndex];
    expect(activeMenu).toHaveClass('color-primary-dark');
  });

  it('renders menu when parent item is active', () => {
    const { getAllByTestId } = render(<VerticalNav expanded={true} menus={menus} active={active} />);

    const childIndex = 2;
    const activeMenu = getAllByTestId('DesignSystem-VerticalNav--Text')[childIndex];
    expect(activeMenu).toHaveClass('color-primary-dark');
  });

  it('renders menu when active prop is updated', () => {
    const { getAllByTestId, rerender } = render(
      <VerticalNav expanded={true} menus={menus} active={{ name: 'patient_360' }} />
    );
    const activeMenu = getAllByTestId('DesignSystem-VerticalNav--Text')[activeIndex];
    expect(activeMenu).toHaveClass('color-primary-dark');

    rerender(<VerticalNav expanded={true} menus={menus} active={active} />);

    const updatedActiveIndex = 2;
    const newActiveMenu = getAllByTestId('DesignSystem-VerticalNav--Text');
    expect(newActiveMenu[updatedActiveIndex]).toHaveClass('color-primary-dark');
    expect(newActiveMenu[activeIndex]).not.toHaveClass('color-primary-dark');
  });
});

describe('Vertical Navigation component prop: onClick', () => {
  it('calls onClick callback', () => {
    const menuClicked = 0;
    const { getAllByTestId } = render(<VerticalNav menus={menus} active={active} onClick={onClick} />);

    const activeMenu = getAllByTestId('DesignSystem-VerticalNav--Item')[menuClicked];
    fireEvent.click(activeMenu);
    expect(onClick).toHaveBeenCalledTimes(1);
    expect(onClick).toHaveBeenCalledWith(menus[menuClicked]);
  });

  it('calls onClick callback if parent item is clicked and panel is collapsed', () => {
    const menuClicked = 1;
    const { getAllByTestId } = render(
      <VerticalNav expanded={false} menus={menus} active={{ name: 'patient_360' }} onClick={onClick} />
    );

    const activeMenu = getAllByTestId('DesignSystem-VerticalNav--Item')[menuClicked];
    fireEvent.click(activeMenu);
    expect(onClick).toHaveBeenCalled();
    expect(onClick).toHaveBeenCalledWith(menus[menuClicked].subMenu![0]);
  });
});

describe('Vertical Navigation component prop: menus with expanded subMenu', () => {
  it("Menu item having expanded state subMenu should be expanded even if it's subMenu is not active", () => {
    const { getAllByTestId } = render(
      <VerticalNav menus={menusWithExpandedSubMenu} active={{ name: 'patient_360' }} onClick={onClick} />
    );

    const timelineSubMenu = getAllByTestId('DesignSystem-VerticalNav--Text')[2];
    expect(timelineSubMenu.textContent).toMatch('Timeline');

    fireEvent.click(timelineSubMenu);

    const careManagement = getAllByTestId('DesignSystem-VerticalNav--Item')[1];
    fireEvent.click(careManagement);

    expect(getAllByTestId('DesignSystem-VerticalNav--Text')[2].textContent).toMatch('Episodes');
  });

  it("Menu item not having expanded state subMenu should be expanded when it's subMenu is active", () => {
    const { getAllByTestId } = render(
      <VerticalNav menus={menus} active={{ name: 'care_management.timeline' }} onClick={onClick} />
    );
    const childItems = getAllByTestId('DesignSystem-VerticalNav--Text');
    expect(childItems[2].textContent).toMatch('Timeline');

    const careManagement = getAllByTestId('DesignSystem-VerticalNav--Item')[1];
    fireEvent.click(careManagement);

    expect(getAllByTestId('DesignSystem-VerticalNav--Text')[2].textContent).toMatch('Episodes');
  });

  it('When subMenu is not active and not in expanded state then it should not be expanded', () => {
    const { getAllByTestId } = render(<VerticalNav menus={menus} active={{ name: 'patient_360' }} onClick={onClick} />);

    const childItems = getAllByTestId('DesignSystem-VerticalNav--Text');
    expect(childItems[2].textContent).toMatch('Episodes');

    const careManagement = getAllByTestId('DesignSystem-VerticalNav--Item')[1];
    fireEvent.click(careManagement);

    expect(getAllByTestId('DesignSystem-VerticalNav--Text')[2].textContent).toMatch('Timeline');
  });
});
