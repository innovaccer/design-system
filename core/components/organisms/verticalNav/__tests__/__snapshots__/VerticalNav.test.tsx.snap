// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vertical Navigation component 
  expanded: false, menus: [{"name":"patient_360","label":"Patient 360","icon":"assignment_ind","link":"/patient360","count":10,"group":"Section 1"},{"name":"care_management","label":"Care Management and Resources","icon":"forum","group":"Section 2","subMenu":[{"name":"care_management.timeline","label":"Timeline","icon":"assignment_ind"},{"name":"care_management.care_plans","label":"Care Plans","icon":"events"}]},{"name":"episodes","label":"Episodes","disabled":true,"icon":"airline_seat_flat_angled","group":"Section 2","count":100},{"name":"risk","label":"Risk","icon":"favorite","group":"Section 2","subMenu":[{"name":"risk.timeline","label":"Timeline"},{"name":"risk.care_plans","label":"Care Plans"}]}], active: {"name":"care_management.timeline"}, onClick: "[Function]", onToggle: "[Function]"
 1`] = `
<body>
  <div>
    <div
      class="VerticalNav"
    >
      <a
        class="MenuItem MenuItem--vertical MenuItem--collapsed color-inverse"
        data-test="DesignSystem-VerticalNav--Item"
        href="/patient360"
        tabindex="0"
      >
        <div
          class="d-flex align-items-center overflow-hidden"
        >
          <i
            class="material-icons material-icons-round Icon"
            data-test="DesignSystem-VerticalNav--Icon"
            role="button"
            style="font-size: 16px; width: 16px;"
          >
            assignment_ind_round
          </i>
        </div>
      </a>
      <a
        class="MenuItem MenuItem--vertical MenuItem--collapsed color-inverse"
        data-test="DesignSystem-VerticalNav--Item"
        tabindex="0"
      >
        <div
          class="d-flex align-items-center overflow-hidden"
        >
          <i
            class="material-icons material-icons-round Icon"
            data-test="DesignSystem-VerticalNav--Icon"
            role="button"
            style="font-size: 16px; width: 16px;"
          >
            forum_round
          </i>
        </div>
      </a>
      <a
        class="MenuItem MenuItem--vertical MenuItem--collapsed MenuItem--active color-primary-dark"
        tabindex="0"
      >
        <div
          class="d-flex align-items-center overflow-hidden"
        >
          <i
            class="material-icons material-icons-round Icon"
            data-test="DesignSystem-VerticalNav--Icon"
            role="button"
            style="font-size: 16px; width: 16px;"
          >
            assignment_ind_round
          </i>
        </div>
      </a>
      <a
        class="MenuItem MenuItem--vertical MenuItem--collapsed color-inverse"
        tabindex="0"
      >
        <div
          class="d-flex align-items-center overflow-hidden"
        >
          <i
            class="material-icons material-icons-round Icon"
            data-test="DesignSystem-VerticalNav--Icon"
            role="button"
            style="font-size: 16px; width: 16px;"
          >
            events_round
          </i>
        </div>
      </a>
      <a
        class="MenuItem MenuItem--vertical MenuItem--collapsed MenuItem--disabled color-inverse-lightest"
        data-test="DesignSystem-VerticalNav--Item"
        tabindex="0"
      >
        <div
          class="d-flex align-items-center overflow-hidden"
        >
          <i
            class="material-icons material-icons-round Icon"
            data-test="DesignSystem-VerticalNav--Icon"
            role="button"
            style="font-size: 16px; width: 16px;"
          >
            airline_seat_flat_angled_round
          </i>
        </div>
      </a>
      <a
        class="MenuItem MenuItem--vertical MenuItem--collapsed color-inverse"
        data-test="DesignSystem-VerticalNav--Item"
        tabindex="0"
      >
        <div
          class="d-flex align-items-center overflow-hidden"
        >
          <i
            class="material-icons material-icons-round Icon"
            data-test="DesignSystem-VerticalNav--Icon"
            role="button"
            style="font-size: 16px; width: 16px;"
          >
            favorite_round
          </i>
        </div>
      </a>
    </div>
  </div>
</body>
`;

exports[`Vertical Navigation component 
  expanded: true, rounded: [true,false], menus: [{"name":"patient_360","label":"Patient 360","icon":"assignment_ind","link":"/patient360","count":10,"group":"Section 1"},{"name":"care_management","label":"Care Management and Resources","icon":"forum","group":"Section 2","subMenu":[{"name":"care_management.timeline","label":"Timeline","icon":"assignment_ind"},{"name":"care_management.care_plans","label":"Care Plans","icon":"events"}]},{"name":"episodes","label":"Episodes","disabled":true,"icon":"airline_seat_flat_angled","group":"Section 2","count":100},{"name":"risk","label":"Risk","icon":"favorite","group":"Section 2","subMenu":[{"name":"risk.timeline","label":"Timeline"},{"name":"risk.care_plans","label":"Care Plans"}]}], active: {"name":"care_management.timeline"}, onClick: "[Function]", onToggle: "[Function]"
 1`] = `
<body>
  <div>
    <div
      class="VerticalNav VerticalNav--expanded"
    >
      <div
        class="VerticalNav-section"
      >
        <span
          class="Text Text--subtle Text--strong Text--small"
          data-test="DesignSystem-VerticalNav--Section"
        >
          Section 1
        </span>
      </div>
      <a
        class="MenuItem MenuItem--vertical MenuItem--expanded MenuItem--rounded color-inverse"
        data-test="DesignSystem-VerticalNav--Item"
        href="/patient360"
        tabindex="0"
      >
        <div
          class="d-flex align-items-center overflow-hidden"
        >
          <i
            class="material-icons material-icons-round Icon mr-4"
            data-test="DesignSystem-VerticalNav--Icon"
            role="button"
            style="font-size: 16px; width: 16px;"
          >
            assignment_ind_round
          </i>
          <span
            class="Text Text--regular color-inverse MenuItem-Text"
            data-test="DesignSystem-VerticalNav--Text"
          >
            Patient 360
          </span>
        </div>
        <span
          class="Pills Badge--secondary MenuItem-count"
          data-test="DesignSystem-VerticalNav--Pills"
        >
          10
        </span>
      </a>
      <div
        class="VerticalNav-section VerticalNav-section--border"
      >
        <span
          class="Text Text--subtle Text--strong Text--small"
          data-test="DesignSystem-VerticalNav--Section"
        >
          Section 2
        </span>
      </div>
      <a
        class="MenuItem MenuItem--vertical MenuItem--expanded MenuItem--rounded color-inverse"
        data-test="DesignSystem-VerticalNav--Item"
        tabindex="0"
      >
        <div
          class="d-flex align-items-center overflow-hidden"
        >
          <i
            class="material-icons material-icons-round Icon mr-4"
            data-test="DesignSystem-VerticalNav--Icon"
            role="button"
            style="font-size: 16px; width: 16px;"
          >
            forum_round
          </i>
          <span
            class="Text Text--regular color-inverse MenuItem-Text"
            data-test="DesignSystem-VerticalNav--Text"
          >
            Care Management and Resources
          </span>
        </div>
        <i
          class="material-icons material-icons-round Icon mx-4"
          role="button"
          style="font-size: 16px; width: 16px;"
        >
          keyboard_arrow_up_round
        </i>
      </a>
      <a
        class="MenuItem MenuItem--vertical MenuItem--expanded MenuItem--active MenuItem--subMenu MenuItem--rounded color-primary-dark"
        tabindex="0"
      >
        <div
          class="d-flex align-items-center overflow-hidden"
        >
          <i
            class="material-icons material-icons-round Icon mr-4"
            data-test="DesignSystem-VerticalNav--Icon"
            role="button"
            style="font-size: 16px; width: 16px;"
          >
            assignment_ind_round
          </i>
          <span
            class="Text Text--regular color-primary-dark MenuItem-Text"
            data-test="DesignSystem-VerticalNav--Text"
          >
            Timeline
          </span>
        </div>
      </a>
      <a
        class="MenuItem MenuItem--vertical MenuItem--expanded MenuItem--subMenu MenuItem--rounded color-inverse"
        tabindex="0"
      >
        <div
          class="d-flex align-items-center overflow-hidden"
        >
          <i
            class="material-icons material-icons-round Icon mr-4"
            data-test="DesignSystem-VerticalNav--Icon"
            role="button"
            style="font-size: 16px; width: 16px;"
          >
            events_round
          </i>
          <span
            class="Text Text--regular color-inverse MenuItem-Text"
            data-test="DesignSystem-VerticalNav--Text"
          >
            Care Plans
          </span>
        </div>
      </a>
      <a
        class="MenuItem MenuItem--vertical MenuItem--expanded MenuItem--disabled MenuItem--rounded color-inverse-lightest"
        data-test="DesignSystem-VerticalNav--Item"
        tabindex="0"
      >
        <div
          class="d-flex align-items-center overflow-hidden"
        >
          <i
            class="material-icons material-icons-round Icon mr-4"
            data-test="DesignSystem-VerticalNav--Icon"
            role="button"
            style="font-size: 16px; width: 16px;"
          >
            airline_seat_flat_angled_round
          </i>
          <span
            class="Text Text--regular color-inverse-lightest MenuItem-Text"
            data-test="DesignSystem-VerticalNav--Text"
          >
            Episodes
          </span>
        </div>
        <span
          class="Pills Badge--subtle-secondary MenuItem-count MenuItem-count--disabled"
          data-test="DesignSystem-VerticalNav--Pills"
        >
          99+
        </span>
      </a>
      <a
        class="MenuItem MenuItem--vertical MenuItem--expanded MenuItem--rounded color-inverse"
        data-test="DesignSystem-VerticalNav--Item"
        tabindex="0"
      >
        <div
          class="d-flex align-items-center overflow-hidden"
        >
          <i
            class="material-icons material-icons-round Icon mr-4"
            data-test="DesignSystem-VerticalNav--Icon"
            role="button"
            style="font-size: 16px; width: 16px;"
          >
            favorite_round
          </i>
          <span
            class="Text Text--regular color-inverse MenuItem-Text"
            data-test="DesignSystem-VerticalNav--Text"
          >
            Risk
          </span>
        </div>
        <i
          class="material-icons material-icons-round Icon mx-4"
          role="button"
          style="font-size: 16px; width: 16px;"
        >
          keyboard_arrow_down_round
        </i>
      </a>
    </div>
  </div>
</body>
`;
